src/components/HistoryTables/MessageHistory/Detail/MethodName.tsx

```diff
@@ -0,0 +1,18 @@
+import { Address } from '../../../../generated/graphql'
+import { Badge, Line } from '../../../Layout'
+import { useMethodName } from '../hooks/useMethodName'
+
+export function MethodName({ address, params, methodNum }: MethodNameProps) {
```

Comment:
Separating out this component fixed the issue with useAbi not updating after setting undefined as an initial value

Reply:
I think this might have had something to do with SSR. Moving it into a piece of code that only renders in the browser makes sure window is defined and the right value will be set. The new useLocalStorage hook makes sure the value resaturated when going from SSR to the browser.